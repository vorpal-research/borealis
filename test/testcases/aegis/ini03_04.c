#include <stdlib.h>

int main(void) {
    int *p;
    
    p = malloc(sizeof(int));
    
    if (*p > 0) {
        free(p);
    }
    
    return *p;
}
