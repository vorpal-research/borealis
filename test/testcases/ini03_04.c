//
// Разыменование неинициализированного или нулевого указателя
//
#include <stdlib.h>

int main(void)
{
    int *p;
    
    p = malloc(sizeof(int));
    
    if (*p > 0) // дефект INI-01 (гарантированный => обрубание)
    {
        free(p);
    }
    
    return *p;  // дефект INI-01,
                //        INI-03,
                //        RES-01
}
