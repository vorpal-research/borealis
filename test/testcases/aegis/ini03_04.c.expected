
[
	{
		"location" : 
		{
			"filename" : "test/testcases/aegis/ini03_04.c",
			"loc" : 
			{
				"col" : 9,
				"line" : 8
			}
		},
		"type" : "INI-03"
	},
	{
		"location" : 
		{
			"filename" : "test/testcases/aegis/ini03_04.c",
			"loc" : 
			{
				"col" : 5,
				"line" : 12
			}
		},
		"type" : "INI-03"
	}
]
